src/diff/props.js

```diff
@@ -42,36 +43,21 @@ function setProperty(dom, name, value, oldValue, isSvg) {
 	name = isSvg ? (name==='className' ? 'class' : name) : (name==='class' ? 'className' : name);
 
 	if (name==='style') {
-
-		/* Possible golfing activities for setting styles:
-		 *   - we could just drop String style values. They're not supported in other VDOM libs.
-		 *   - assigning to .style sets .style.cssText - TODO: benchmark this, might not be worth the bytes.
-		 *   - assigning also casts to String, and ignores invalid values. This means assigning an Object clears all styles.
-		 */
-		let s = dom.style;
-
-		if (typeof value==='string') {
-			s.cssText = value;
-		}
-		else {
-			if (typeof oldValue==='string') s.cssText = '';
-			else {
-				// remove values not in the new list
-				for (let i in oldValue) {
-					if (value==null || !(i in value)) s.setProperty(i.replace(CAMEL_REG, '-'), '');
-				}
-			}
-			for (let i in value) {
-				const v = value[i];
-				if (oldValue==null || v!==oldValue[i]) {
-					s.setProperty(i.replace(CAMEL_REG, '-'), typeof v==='number' && IS_NON_DIMENSIONAL.test(i)===false ? (v + 'px') : v);
-				}
+		const set = assign(assign({}, oldValue), value);
+		for (let i in set) {
+			if ((value || EMPTY_OBJ)[i] === (oldValue || EMPTY_OBJ)[i]) {
+				continue;
 			}
+			dom.style.setProperty(
+				(i[0] === '-' && i[1] === '-') ? i : i.replace(CAMEL_REG, '-$&'),
```

Comment:
Not sure about the performance impact, but this is short: `/^--/.test(i)`

I made a ESBench, but it's all over the place though: https://esbench.com/bench/5cd320f74cd7e6009ef623bf

Reply:
Cool, that does cut down preact.js.gz size down by 3 bytes indeed! Changing the benchmark from `text += "match"` to something like `count += 1` (with `let count = 0;` in the setup) stabilized the results to something like this (Chrome 74, MacOS 10.14.4):
<img width="542" alt="Screenshot 2019-05-08 at 22 23 32" src="https://user-images.githubusercontent.com/19776768/57402429-feeb7400-71df-11e9-84f0-855c62901d3c.png">

Reply:
Hey @afeiship yeah, it compresses better I guess, but "direct match" is waaay faster 😄 and we might get to save 3B from somewhere else.

Reply:
@jviide nice thanks for the analysis.

I had made tests across the browser but I had so highly varying results (sometimes the precompiled was even faster than direct.. 🙈 ) so I wasn't sure about how much of a performance hit it is (apparently a brutal one 😄 ), just wanted to put it up for discussion. 😅

Reply:
Much appreciated - keep 'em coming! 👍 I was pretty surprised how big the performance difference was (and there's always the possibility that I botched the benchmark).

@marvinhagemeister pointed out that `setProperty` in `props.js` uses the same approach to identify event handlers, based on a similar benchmark: https://github.com/developit/preact/blob/7e53dd29003bbbddf5bdbc934c8e00c69b57936c/src/diff/props.js#L75-L76